src/main/java/org/veupathdb/service/eda/ss/model/db/FilteredResultFactory.java

```diff
@@ -261,6 +377,20 @@ private static List<String> getDateVarNames(Entity entity, List<String> desiredV
         .collect(Collectors.toList());
   }
 
+  static Iterator<Map<String, String>> iteratorFromWideResult(Iterator<Map<String, String>> tallRowsIterator,
+                                     Entity outputEntity, List<String> headers) {
+    // an iterator of lists of maps, each list being the rows of the tall table returned for a single entity id
```

Comment:
This comment seems to conflict with the name of this method.  Is the incoming iterator tall rows or wide rows?

Reply:
Ok, based on the two conversions here, looks like it is the tall-to-wide conversion w/tall row grouping.  So maybe a few name changes to clarify but logic seems ok.